infra/gcp/bash/prow/ensure-e2e-projects.sh

```diff
@@ -112,6 +113,17 @@ function ensure_e2e_project() {
       "serviceAccount:${PROW_BUILD_SVCACCT}" \
       "roles/iam.serviceAccountUser"
 
+    # Ensure GCP Default Compute Service Account can administer Secret Manager secrets
+    ensure_project_role_binding "${prj}" \
+      "serviceAccount:${PROW_BUILD_SVCACCT}" \
+      "roles/secretmanager.admin"
+
+    # Ensure GCP Default Compute Engine Service Agent Account can manage Secret Manager Secrets
+    ensure_project_role_binding "${prj}" \
+      "serviceAccount:service-${project_number}@compute-system.iam.gserviceaccount.com" \
+      "roles/secretmanager.admin"
```

Comment:
In which case we want the default service agent to create secrets ? do we have tests for this ?

Reply:
https://github.com/dargudear-google/test-infra/blob/master/config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml#L682-L725
This prow job will trigger test https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/test/bats/gcp.bats

Currently this test, access secrets (Google Secret Manager) from google managed projects, but now we want to remove this dependency as everything should be community owned.

ref: https://github.com/kubernetes/k8s.io/issues/7246#issuecomment-2344972240

Reply:
Yes, you are correct. Thanks!
Removed the config for compute service agent.